docs/sphinx/user_guide/feature/reduction.rst

```diff
@@ -236,38 +240,85 @@ RAJA::expt::Reduce
           compatible with the ``EXEC_POL``. ``Seg`` is the iteration space
           object for ``RAJA::forall``.
 
-.. important:: The order and types of the local reduction variables in the
-               kernel body lambda expression must match exactly with the
-               corresponding ``RAJA::expt::Reduce`` arguments to the
-               ``RAJA::forall`` to ensure that the correct result is obtained.
+.. important:: The local reduction arguments to the lambda expression must be
+               ``RAJA::expt::ValOp`` references. Each ``ValOp`` references
+               corresponds to a ``RAJA::expt::Reduce`` call within the forall
+               arguments. The ``ValOp`` reduction data type and RAJA operator need
+               to match the data type referenced and operator template argument
+               in the ``RAJA::expt::Reduce`` call. Finally, the ordering of the
+               ``ValOp`` references must correspond to the ordering of the
+               ``RAJA::expt::Reduce`` calls to ensure that the correct result is
+               obtained.
 
 RAJA::expt::ValLoc
 ..................
 
 As with the current RAJA reduction interface, the new interface supports *loc*
 reductions, which provide the ability to get a kernel/loop index at which the
 final reduction value was found. With this new interface, *loc* reductions
-are performed using ``ValLoc<T>`` types. Since they are strongly typed, they
-provide ``min()`` and ``max()`` operations that are equivalent to using
-``RAJA_MIN()`` or ``RAJA_MAX`` macros as demonstrated in the code example below.
-Users must use the ``getVal()`` and ``getLoc()`` methods to access the reduction
-results::
+are performed using ``ValLoc<T,I>`` types, where ``T`` is the underlying data type,
+and ``I`` is the index type. Users must use the ``getVal()`` and ``getLoc()``
+methods to access the reduction results.
+
+In the kernel body lambda expression, a ``ValLoc<T,I>`` must be wrapped in a
```

Comment:
```suggestion
A ``ValLoc<T,I>`` must be wrapped in a
```

The first part of that sentence is confusing and clouds the real point of the rest of the sentence.

Reply:
I think we need to keep the first part of the sentence because there is a difference between the user-declared `ValLoc`, and the `ValLoc` used within the lambda (which needs to be wrapped).

This whole thing is required because of the exposure of the `ValOp<ValLoc<>>` type, and I have an idea to hide this from the user, but we can address this in another issue or PR.